### PR TITLE
aws - add tag and remove-tag actions to roles

### PR DESCRIPTION
**Pull request by coingraham**
WIP - the actual actions are working fine but I can't get the filters to work, nor the first assert to pass because the Tags aren't being returned in the list_roles call.

### CODE REVIEW

**Comment by coingraham** [comment]
Attempt to fix for #3160

**Comment by coingraham** [comment]
Weird, I'm not getting tags back from Boto3 or the AWS Cli.  I do some more testing tomorrow.

File: c7n/resources/iam.py [hunk 1]
```diff
@@ -104,6 +104,42 @@ def get_resources(self, resource_ids, cache=True):
...
+    def process_resource_set(self, client, roles, tags):
+        for role in roles:
+            try:
+                client.untag_role(RoleName=role['RoleName'], TagKeys=tags)
+            except client.exceptions.NoSuchEntityException:
+                continue
+
+
+@Role.action_registry.register('mark-for-op')
+class RoleTagDelayedAction(TagDelayedAction):
```
Comment (kapilt) [inline c7n/resources/iam.py]: instead of defining an empty class could be registered as just a function per marked-for-op below

Comment (kapilt) [reply c7n/resources/iam.py]: the tag machinery was refactored since the iam user tagging support, to make subclasses of mark-for-op extraneous.

**Comment by kapilt** [comment]
re not getting tags back from boto or cli, that would imply needing to upgrade on them. also this pr won't work without alot more surgery on previous tests etc, as tags need to be augmented onto the role.

**Comment by kapilt** [comment]
ah i see your confusion, the docs are wrong btw, list roles doesn't return tags.

**Comment by kapilt** [comment]
@coingraham fwiw, on trunk (merged into this branch) roles return tags the way you would expect.

**Comment by coingraham** [comment]
Finally got that working with the tags.  Removed the mark-to-op.

**Comment by coingraham** [comment]
Looks like tox checks are failing on an unrelated test missing placebo data.  Can you take a look @kapilt ?

**Comment by coingraham** [comment]
I added the file that the check was looking for.

**Pull request merged**